app/Http/Controllers/Api/RuleController.php

```diff
@@ -0,0 +1,229 @@
+     * @apiParam {String} rule The rule which should be in the format %entity $condition $value (i.e `%devices.status != 0` for devices marked as down).
+     * @apiParam {Boolean} [mute=false] Show alert status but mute notifications
+     * @apiParam {Boolean} [disabled=false] Disable rule
+     * @apiParam {String="ok","critical","warning"} severity How to display the alert, OK:`Green`, Warning:`Yellow`, Critical:`Red`
+     * @apiParam {Boolean} [invert=false] Alert when this rule doesn't match
+     * @apiParam {Number} [count=-1] This is how many polling runs before an alert will trigger and the frequency.
+     * @apiParam {Array} [devices] An array of device ID's to map to this rule. If empty the rule will be global.
+     *
+     * @apiSuccessExample Success-Response:
+     *     HTTP/1.1 200 OK
+     *     {
+     *          "data":
+     *          {
+     *              "id": 1,
+     *              "rule": "%macros.device_down = \"1\"",
+     *              "severity": "critical",
+     *              "extra": "{\"mute\":false,\"count\":-1,\"delay\":300,\"invert\":false,\"interval\":300}",
+     *              "disabled": "0",
+     *              "name": "Devices up/down",
+     *              "query": "SELECT * FROM devices WHERE (devices.device_id = ?) && (((devices.status = 0  &&  ((devices.disabled = 0  &&  devices.ignore = 0)))) = \"1\")",
+     *              "builder": "",
+     *              "proc": null
+     *          },
+     *      }
+     *
+     * @apiErrorExample {json} Error-Response:
+     *      HTTP/1.1 422 Unproccesssable Entity
+     *      {
+     *          "rule": [
+     *              "The rule field is required."
+     *          ],
+     *          "name": [
+     *              "The name field is required."
+     *          ]
+     *      }
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
+            'name'  =>  'required|unique:alert_rules,name',
+            'mute'  =>  'ext_bool',
+            'disabled'  => 'ext_bool',
+            'severity'  => [\Illuminate\Validation\Rule::in(['ok', 'critical', 'warning']), 'required'],
+            'invert'    => 'ext_bool',
+            'count'     => 'numeric',
+            'rule'      => 'required',
+            'devices'   => 'array'
+        ]);
+
+        $extra = [
+            'count' =>  $request->get('count', -1),
+            'mute'  =>  $request->get('mute', false),
+            'delay' =>  $request->get('delay', 300),
+            'invert' => $request->get('invert', false),
+            'interval' => $request->get('interval', 300)
+        ];
+
+        $rule = Rule::create([
+            'name'      => $request->name,
+            'disabled'  => $request->get('disabled', false),
+            'severity'  => $request->severity,
+            'rule'      => $request->rule,
+            'extra'     => json_encode($extra),
+            'query'     => '' // TODO: Actually generate query?
```

Comment:
@laf Where / wow is this generated?

Reply:
We auto generate it from the 'builder' column. See `html/includes/forms/alert-rules.inc.php`